File: backend/routers/image.py
from fastapi import APIRouter, UploadFile, File, Form
from models.schemas import ImageQueryResponse
from services.gemini_service import analyze_image

router = APIRouter(prefix="/api/image", tags=["image"])


@router.post("/query", response_model=ImageQueryResponse)
async def query_image(
    file: UploadFile = File(...),
    prompt: str = Form(default="Describe this image in detail."),
):
    image_bytes = await file.read()
    answer = analyze_image(image_bytes, prompt)
    return ImageQueryResponse(answer=answer)

File: frontend/src/api/image.js
import client from "./client";

export const queryImage = (file, prompt = "") => {
  const form = new FormData();
  form.append("file", file);
  form.append("prompt", prompt || "Describe this image in detail.");
  return client.post("/api/image/query", form).then((r) => r.data.answer);
};

export const generateImage = (prompt) =>
  client.post("/api/generate/image", { prompt }).then((r) => r.data.image_base64);
